lib/prototypes/abstract.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const { formatters, checks, harmonize } = require('../util.js');
+
+const AbstractType = {
+  formatters: {},
+  checks: {},
+  check(value, path, root) {
+    try {
+      const typeErr = harmonize(this.checkType(value, path, root));
+      if (typeErr) return typeErr;
+      if (this.validate) {
+        const validErr = harmonize(this.validate(value, path, root));
+        if (validErr) return validErr;
+      }
+      for (const [name, subCheck] of Object.entries(this.checks)) {
```

Comment:
Why not have `checks` as Map or even an Array then?

Reply:
It never will be large, so no reason to use map. But I will try an array

Reply:
Changed to an array

Reply:
Impossible to use array, because field may not use `subCheck` example: `{ type: 'string' }` may not use length at all. `{ type: 'string', length: 10 }` may use length `subCheck`

Reply:
Answered